q2_composition/_ancombc.py

```diff
@@ -120,6 +120,25 @@ def _ancombc(table, metadata, formula, p_adj_method, prv_cut, lib_cut,
                                  ' \n\n'
                                  ' column::value pair with a value that was'
                                  ' not found: "%s"' % i)
+
+            # check that IDs associated with chosen reference level(s) are
+            # present within the input table
+            level_value_idx = meta.index[meta[column] == level_value].tolist()
+            table_idx_list = table.index.tolist()
+
+            if not list(set(level_value_idx) & set(table_idx_list)):
```

Comment:
Pandas index objects implement an efficient intersect which is going to be better building two sets out of iteration, since you already have the index.